### PR TITLE
Update UG and fix minor bugs

### LINKED ISSUE #121: Update UG
* intro
* budget
* overview
* balance
* exit
* saving data
* command summary for the above
* table of contents

### PR DESCRIPTION
**Pull request by ryan1604**
Fixes #121

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -48,6 +48,17 @@ API: `Storage.java`
 * The `load` method in LoadData reads the `data.txt` file and loads any existing Income, Expense and Budget into the application.
 * The `save` method in SaveData saves all Incomes, Expenses and existing Budget into the `data.txt` file.
 
+#### Design considerations:
+
+* There are 2 main ways to implement the storage, one is to save the data after every command, and the other is to save 
+the data one upon exiting the program with the `exit` command.
+* Saving the data once upon exit (Currently implemented):
+  * Advantage: Better efficiency and performance of the program.
+  * Disadvantage: If the program crashes or is exit incorrectly, data will not be saved.
```
Comment (NeoMinWei) [inline docs/DeveloperGuide.md]: "exited" !!!!!!!!!
or "crashes or exits" !!!!

**Review by NeoMinWei — APPROVED** [review]
LGTM

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (ryan1604) [reply docs/DeveloperGuide.md]: 💀 i changed it to the second one

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -48,6 +48,17 @@ API: `Storage.java`
 * The `load` method in LoadData reads the `data.txt` file and loads any existing Income, Expense and Budget into the application.
 * The `save` method in SaveData saves all Incomes, Expenses and existing Budget into the `data.txt` file.
 
+#### Design considerations:
```
Comment (wwweert123) [inline docs/DeveloperGuide.md:51]: Good clarification of our design consideration

File: docs/UserGuide.md [hunk 3]
```diff
@@ -2,7 +2,9 @@
 
 ## Introduction
 
-{Give a product intro}
+Financial Planner is a Command Line Interface (CLI) application for managing your finances conveniently.
```
Comment (wwweert123) [inline docs/UserGuide.md:24]: UserGuide looks very good!

Shall we add a table of contents?

File: docs/diagrams/Storage.puml [hunk 4]
```diff
@@ -8,8 +8,8 @@ Class Storage
 Class "{abstract}\nLoadData" as LoadData
 Class "{abstract}\nSaveData" as SaveData
 Class CashflowList
-Class Budget
-Class Cashflow
+Class "{abstract}\nBudget" as Budget
```
Comment (wwweert123) [inline docs/diagrams/Storage.puml:11]: 👍

**Review by wwweert123 — APPROVED** [review]
Very good!

File: docs/UserGuide.md (hunk 3 above)
Comment (ryan1604) [reply docs/UserGuide.md:24]: added!

**Pull request merged**